src/Compiler/Microsoft.AspNetCore.Razor.Language/src/BuilderExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor.PooledObjects;
+using Microsoft.Extensions.ObjectPool;
+
+namespace Microsoft.AspNetCore.Razor.Language;
+
+internal static class BuilderExtensions
+{
+    public static T[] BuildAll<T, TBuilder>(this List<TBuilder> builders, ObjectPool<HashSet<T>> pool)
+        where TBuilder : IBuilder<T>
+    {
+        using var set = new PooledHashSet<T>(pool);
+
+        foreach (var builder in builders)
+        {
+            set.Add(builder.Build());
+        }
+
+        return set.ToArray();
```

Comment:
Should we add a doc comment mentioning the return here is unordered given the input is ordered?

Reply:
Sure, I'll add a doc comment. Note that this is the same behavior that was already present.

Reply:
Actually, I went ahead and made the result ordered. :smile: